plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorTest.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    /**
+     * DataProvider overload to exclude 2 test cases from generic tests:
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnCtas(Session, String, int)}
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnInsert(Session, String, int)}
+     * <p>
+     * 1. partitioning -> 'bucket(custkey, 4)', 'truncate(comment, 1)'
+     * Test environment provides higher memory usage reaching over 1,5GB per test case.
+     * This leads to test flakiness when both test classes run in parallel and may consume
+     * over 3GB which is current hard limit for CI and may end up with OOM.
+     * <p>
+     * Limiting dataset to 1500 rows in dedicated test methods for MinIO solves this issue:
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnCtasPartitionComplex()}
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnInsertPartitionComplex()}
+     * <p>
+     * 2. partitioning -> 'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'
+     * Test environment causes HMS to consume more time during createTable operation.
+     * This leads to read timed out and retry operation ended with TableAlreadyExists
+     * (HMS managed to end operation on it's side). As such behaviour wasn't observed
+     * in normal circumstances, test was adopted to avoid it.
+     * <p>
+     * Limiting dataset to 300 rows in dedicated test methods for MinIO solves this issue:
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnCtasPartitionWithSameColumnMultipleTimes()}
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnInsertPartitionWithSameColumnMultipleTimes()}
+     */
+    @Override
+    @DataProvider
+    public Object[][] repartitioningDataProvider()
+    {
+        Session obeyConnectorPartitioning = Session.builder(getSession())
+                .setSystemProperty(PREFERRED_WRITE_PARTITIONING_MIN_NUMBER_OF_PARTITIONS, "1")
+                .build();
+        return new Object[][] {
+                // identity partitioning column
+                {obeyConnectorPartitioning, "'orderstatus'", 3},
+                // bucketing
+                {getSession(), "'bucket(custkey, 13)'", 13},
+                // varchar-based
+                {getSession(), "'truncate(comment, 1)'", 35}
+        };
+    }
+
+    @Test
+    @Override
+    public void testShowCreateTable()
+    {
+        assertThat((String) computeScalar("SHOW CREATE TABLE region"))
+                .isEqualTo("" +
+                        "CREATE TABLE iceberg.tpch.region (\n" +
+                        "   regionkey bigint,\n" +
+                        "   name varchar,\n" +
+                        "   comment varchar\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   format = '" + format + "',\n" +
+                        "   location = 's3://" + bucketName + "/tpch/region'\n" +
```

Comment:
The table was created without location, or with location?

Reply:
It was created/cloned by `io.trino.testing.QueryAssertions#copyTable)` with CTAS from `tpch.tiny.region` during test setUp